### DISCUSSION TITLE
Multiple users username mixing up

### DISCUSSION THREAD

**Discussion by Giann8** [telegraf/telegraf#1849] [Q&A] [answered]
Hello everyone, I’m programming a gamebot with missions, it should run forever but the problem is when there are more than one users. if two or more users start the bot the username (tried to store it in a map with chat_id as a key) still remains the username of the last or the first one, so if the second starts a mission the bot doesn’t send the message to the second user but to the first (the obtained object is stored in the inventory of the wrogn username). the map variable is global.
```
bot.start(async (ctx) => {
    var User = new Character
    User.usern = ctx.message.from.username

    let GetId = () => { return ctx.chat.id }

    if (User.usern == null) {
        return ctx.sendMessage('A quanto pare non hai impostato alcuno username, perfavore vai nelle impostazioni di telegram e crea un nuovo username poi torna e utilizza /start')
    }

    let Player = await dbo.findOne({ usern: User.usern })
    if (Player == null) {
        dbo.updateOne(
            { usern: User.usern },
            { $set: { usern: User.usern, name: ctx.message.from.first_name, level: User.level, exp: User.exp, backpack: User.backpack } },
            { upsert: true }
        )
        return ctx.sendMessage('Hey benvenuto, a quanto pare sei nuovo restarta il bot prima di continuare con /start')
    }
    User = Player

    InGameUsers.set(GetId, User)

    ctx.scene.enter('Università_id')

```

**Comment by MKRhere** [comment]
```TS
{ return ctx.chat.id }
```

You're using the same chat ID for all players in a chat. Use `ctx.from.id` or combine the two to make it unique per chat per user.

**Reply by Giann8** [reply]
I resolved using ctx.message.from.id, ho does it changes from user.id?

**Reply by MKRhere** [reply, marked as answer]
I made a mistake. It is `ctx.from.id` (shorthand for `ctx.message.from.id`)
